### ISSUE TITLE
Store special test cases as separate files

### ISSUE THREAD

**Issue by elfring** [namhyung/uftrace#193]
[A few test cases](https://github.com/namhyung/uftrace/blob/55ee3007c44ff1c833b166ff5256201011137f66/uftrace.c#L1024 "Specification of test cases") are also stored in the source file for the main program.
Would it be a bit better to store them as separate source files so that they will be included only on concrete demand?

**Comment by namhyung** [comment]
I think it's better keeping the unit test together (unless it's too long) to help see the definition and usage at once.  They are separated using a preprocessor macro guard.  Also it needs to use static functions - oh did you mean something like `#include "xxx-test.c"`?

**Comment by elfring** [comment]
>… - oh did you mean something like `#include "xxx-test.c"`?

Yes. — If you would insist to keep such source code there at all for example.

**Comment by namhyung** [comment]
Thanks so much to take your time reviewing the source code in detail.  I'll consider separating unit test files once I have enough test cases.

**Comment by elfring** [comment]
I suggest to apply another software refactoring before questionable code will be preserved at inappropriate places.

**Comment by namhyung** [comment]
What do you mean by "another software refactoring"?

**Comment by elfring** [comment]
[Example](https://refactoring.com/catalog/ "Catalog of refactorings"):
Under which circumstances will a bit of source code be moved into another file?

**Comment by namhyung** [comment]
I don't have a firm criteria on it but I feel hard to read when a source code gets too long (maybe 2000+ loc?)

**Comment by elfring** [comment]
I propose to extract mentioned test cases earlier.

**Comment by namhyung** [comment]
I'll put it into the TODO list. :)

**Comment by SEOKMIN83** [comment]
Hi, If this issue is still valid, can I try it?

**Comment by elfring** [comment]
:thought_balloon: Probably, yes.
